src/lib/parameters/usr_parameters_if.cpp

```diff
@@ -0,0 +1,220 @@
+{
+	paramiocget_t data = {param, false, val, PX4_ERROR};
+	boardctl(PARAMIOCGET, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+int
+param_get_default_value(param_t param, void *default_val)
+{
+	paramiocget_t data = {param, true, default_val, PX4_ERROR};
+	boardctl(PARAMIOCGET, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+void
+param_control_autosave(bool enable)
+{
+	paramiocautosave_t data = {enable};
+	boardctl(PARAMIOCAUTOSAVE, reinterpret_cast<unsigned long>(&data));
+}
+
+int param_set(param_t param, const void *val)
+{
+	paramiocset_t data = {param, true, val, PX4_ERROR};
+	boardctl(PARAMIOCSET, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+int param_set_no_notification(param_t param, const void *val)
+{
+	paramiocset_t data = {param, false, val, PX4_ERROR};
+	boardctl(PARAMIOCSET, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+bool param_used(param_t param)
+{
+	paramiocused_t data = {param, false};
+	boardctl(PARAMIOCUSED, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+void param_set_used(param_t param)
+{
+	paramiocsetused_t data = {param};
+	boardctl(PARAMIOCSETUSED, reinterpret_cast<unsigned long>(&data));
+}
+
+int param_set_default_value(param_t param, const void *val)
+{
+	paramiocsetdefault_t data = {param, val, PX4_ERROR};
+	boardctl(PARAMIOCSETDEFAULT, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+int param_reset(param_t param)
+{
+	paramiocreset_t data = {param, true, PX4_ERROR};
+	boardctl(PARAMIOCRESET, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+int param_reset_no_notification(param_t param)
+{
+	paramiocreset_t data = {param, false, PX4_ERROR};
+	boardctl(PARAMIOCRESET, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+void
+param_reset_all()
+{
+	paramiocresetgroup_t data = {PARAM_RESET_ALL, nullptr, 0};
+	boardctl(PARAMIOCRESETGROUP, reinterpret_cast<unsigned long>(&data));
+}
+
+void
+param_reset_excludes(const char *excludes[], int num_excludes)
+{
+	paramiocresetgroup_t data = {PARAM_RESET_EXCLUDES, excludes, num_excludes};
+	boardctl(PARAMIOCRESETGROUP, reinterpret_cast<unsigned long>(&data));
+}
+
+void
+param_reset_specific(const char *resets[], int num_resets)
+{
+	paramiocresetgroup_t data = {PARAM_RESET_SPECIFIC, resets, num_resets};
+	boardctl(PARAMIOCRESETGROUP, reinterpret_cast<unsigned long>(&data));
+}
+
+int param_save_default()
+{
+	paramiocsavedefault_t data = {PX4_ERROR};
+	boardctl(PARAMIOCSAVEDEFAULT, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+int
+param_load_default()
+{
+	paramiocloaddefault_t data = {PX4_ERROR};
+	boardctl(PARAMIOCLOADDEFAULT, reinterpret_cast<unsigned long>(&data));
+	return data.ret;
+}
+
+int
+param_export(const char *filename, param_filter_func filter)
+{
+	paramiocexport_t data = {filename, PX4_ERROR};
+
+	if (filter) { PX4_ERR("ERROR: filter not supported in userside blob"); }
```

Comment:
Note that this is used for factory calibration: https://github.com/PX4/PX4-Autopilot/blob/master/src%2Fmodules%2Fcommander%2Ffactory_calibration_storage.cpp#L83 (not immediately relevant)

Reply:
Yes; if the fileter is needed in user side, I need to come back to this. I admit that I just took a shortcut here and added the ERROR to remind if this pops up...